express-notes-api/server.js

```diff
@@ -0,0 +1,116 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+const middle = express.json();
+
+app.use(middle);
+
+async function reader() {
+  try {
+    let data = await readFile('data.json', 'utf8');
+    data = JSON.parse(data);
+    return data;
+  } catch (error) {
+    console.log('Error: ', error);
+  }
+}
+
+async function writer(data, res) {
+  try {
+    await writeFile('data.json', JSON.stringify(data, null, 2));
+  } catch (error) {
+    const err = {
+      error: 'An unexpected error occurred.',
+    };
+    console.error(error);
+    res.status(500).json(err);
+  }
```

Comment:
Same with this try/catch, though at least here you are returning a 500 to the caller!